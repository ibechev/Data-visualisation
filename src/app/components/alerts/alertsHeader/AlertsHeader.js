import React from "react";
import PropTypes from "prop-types";

import Button from "../../button/Button";

const AlertsHeader = ({
  loading,
  latestAlerts,
  handleSave,
  handleImportance,
  handleChart,
  handleFilter,
  handleUndo,
  handleSettings
}) => {
  return (
    <header className="alerts-header">
      <div className="alerts-info">
        <h4 className="alerts-title">Alerts</h4>
        {!loading && (
          <p className="alerts-latest">Latest alerts ({latestAlerts})</p>
        )}
      </div>
      <div className="alerts-controls">
        <div className="main-alerts-controls">
          <ul className="controls-list">
            <li className="control-li">
              <Button noBody onClick={handleChart}>
                <i className="fas fa-chart-pie" />
              </Button>
            </li>
            <li className="control-li">
              <Button noBody onClick={handleFilter}>
                <i className="fas fa-filter" />
              </Button>
            </li>
            <li className="control-li">
              <Button noBody onClick={handleUndo}>
                <i className="fas fa-undo-alt" />
              </Button>
            </li>
            <li className="control-li">
              <Button noBody onClick={handleSettings}>
                <i className="fas fa-cog" />
              </Button>
            </li>
          </ul>
          <Button onClick={handleSave}>
            Save <i className="fas fa-caret-down" />
          </Button>
        </div>

        {!loading && (
          <div className="sort-alerts-controls">
            <Button noBody onClick={handleImportance}>
              IMPORTANCE <i className="fas fa-caret-down" />
            </Button>
          </div>
        )}
      </div>
    </header>
  );
};

AlertsHeader.propTypes = {
  loading: PropTypes.bool.isRequired,
  latestAlerts: PropTypes.number,
  handleSave: PropTypes.func.isRequired,
  handleImportance: PropTypes.func.isRequired,
  handleChart: PropTypes.func.isRequired,
  handleFilter: PropTypes.func.isRequired,
  handleUndo: PropTypes.func.isRequired,
  handleSettings: PropTypes.func.isRequired
};

export default AlertsHeader;
