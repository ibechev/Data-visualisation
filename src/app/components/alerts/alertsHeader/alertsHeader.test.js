import React from "react";
import { mount } from "enzyme";
import AlertsHeader from "./AlertsHeader";
import renderer from "react-test-renderer";
import { propTypeErrors } from "../../../../../tests/config/testUtils";

describe("Expect <AlertsHeader />", () => {
  const defaultProps = {
    loading: true,
    error: false,
    latestAlerts: 10,
    handleSave: () => {},
    handleImportance: () => {},
    handleChart: () => {},
    handleFilter: () => {},
    handleUndo: () => {},
    handleSettings: () => {}
  };

  const setup = (testProps = {}) => {
    const props = {
      ...defaultProps,
      ...testProps
    };
    return mount(<AlertsHeader {...props} />);
  };

  it("to render correctly", () => {
    const wrapper = renderer
      .create(<AlertsHeader {...defaultProps} />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("to not throw error with corrent props", () => {
    propTypeErrors(<AlertsHeader {...defaultProps} />, defaultProps);
  });

  describe("when prop 'loading' is 'true' and prop 'error' is 'false'", () => {
    const wrapper = setup();

    it("to not display <p class='alerts-latest'", () => {
      expect(wrapper.find("p.alerts-latest")).toHaveLength(0);
    });

    it("to not display <div class='sort-alerts-controls'", () => {
      expect(wrapper.find("div.sort-alerts-controls")).toHaveLength(0);
    });
  });

  describe("when prop 'loading' is 'false' and prop 'error' is 'true'", () => {
    const testProps = {
      loading: false,
      error: true
    };
    const wrapper = setup(testProps);

    it("to not display <p class='alerts-latest'", () => {
      expect(wrapper.find("p.alerts-latest")).toHaveLength(0);
    });

    it("to not display <div class='sort-alerts-controls'", () => {
      expect(wrapper.find("div.sort-alerts-controls")).toHaveLength(0);
    });
  });

  describe("when prop 'loading' is 'true' and prop 'error' is 'true'", () => {
    const testProps = {
      loading: true,
      error: true
    };
    const wrapper = setup(testProps);

    it("to not display <p class='alerts-latest'", () => {
      expect(wrapper.find("p.alerts-latest")).toHaveLength(0);
    });

    it("to not display <div class='sort-alerts-controls'", () => {
      expect(wrapper.find("div.sort-alerts-controls")).toHaveLength(0);
    });
  });

  describe("when prop 'loading' is 'false' and prop 'error' is 'false'", () => {
    const testProps = {
      loading: false,
      error: false
    };
    const wrapper = setup(testProps);

    it("to display <p class='alerts-latest'", () => {
      expect(wrapper.find("p.alerts-latest")).toHaveLength(1);
    });

    it("to display <div class='sort-alerts-controls'", () => {
      expect(wrapper.find("div.sort-alerts-controls")).toHaveLength(1);
    });
  });

  it("to call button handlers on each click of a button", () => {
    const spyBtnHandler = jest.fn();
    const testProps = {
      handleSave: spyBtnHandler,
      handleImportance: spyBtnHandler,
      handleChart: spyBtnHandler,
      handleFilter: spyBtnHandler,
      handleUndo: spyBtnHandler,
      handleSettings: spyBtnHandler,
      loading: false
    };
    const wrapper = setup(testProps);

    wrapper.find("button").forEach(button => button.simulate("click"));

    expect(spyBtnHandler).toHaveBeenCalledTimes(wrapper.find("button").length);
  });

  it("to display the corrent amount of latest alerts", () => {
    const wrapper = setup({ loading: false });
    expect(wrapper.find(".alerts-latest").text()).toEqual(
      `Latest alerts (${defaultProps.latestAlerts})`
    );
  });
});
