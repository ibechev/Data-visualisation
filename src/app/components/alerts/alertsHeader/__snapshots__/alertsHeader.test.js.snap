// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <AlertsHeader /> to render correctly 1`] = `
<header
  className="alerts-header"
>
  <div
    className="alerts-info"
  >
    <h4
      className="alerts-title"
    >
      Alerts
    </h4>
  </div>
  <div
    className="alerts-controls"
  >
    <div
      className="main-alerts-controls"
    >
      <ul
        className="controls-list"
      >
        <li
          className="control-li"
        >
          <button
            className="button no-body"
            onClick={[Function]}
            type="button"
          >
            <i
              className="fas fa-chart-pie"
            />
          </button>
        </li>
        <li
          className="control-li"
        >
          <button
            className="button no-body"
            onClick={[Function]}
            type="button"
          >
            <i
              className="fas fa-filter"
            />
          </button>
        </li>
        <li
          className="control-li"
        >
          <button
            className="button no-body"
            onClick={[Function]}
            type="button"
          >
            <i
              className="fas fa-undo-alt"
            />
          </button>
        </li>
        <li
          className="control-li"
        >
          <button
            className="button no-body"
            onClick={[Function]}
            type="button"
          >
            <i
              className="fas fa-cog"
            />
          </button>
        </li>
      </ul>
      <button
        className="button primary"
        onClick={[Function]}
        type="button"
      >
        Save 
        <i
          className="fas fa-caret-down"
        />
      </button>
    </div>
  </div>
</header>
`;
